src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/categories/CategorySyncIT.java

```diff
@@ -0,0 +1,496 @@
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+
+import static com.commercetools.sync.commons.utils.SyncUtils.batchDrafts;
+import static com.commercetools.sync.commons.utils.SyncUtils.replaceCategoriesReferenceIdsWithKeys;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.OLD_CATEGORY_CUSTOM_TYPE_KEY;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.createCategories;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.createCategoriesCustomType;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.createChildren;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.deleteAllCategories;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.getCategoryDrafts;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.getCategoryDraftsWithPrefix;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.getCustomFieldsJsons;
+import static com.commercetools.sync.integration.commons.utils.CategoryITUtils.syncBatches;
+import static com.commercetools.sync.integration.commons.utils.ITUtils.deleteTypesFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.ITUtils.getStatisticsAsJSONString;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_SOURCE_CLIENT;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CategorySyncIT {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CategorySyncIT.class);
+    private CategorySync categorySync;
+
+    private List<String> callBackErrorResponses = new ArrayList<>();
+    private List<Throwable> callBackExceptions = new ArrayList<>();
+    private List<String> callBackWarningResponses = new ArrayList<>();
+
+    /**
+     * Delete all categories and types from source and target project. Then create custom types for source and target
+     * CTP project categories.
+     */
+    @BeforeClass
+    public static void setup() {
+        deleteAllCategories(CTP_TARGET_CLIENT);
+        deleteAllCategories(CTP_SOURCE_CLIENT);
+        deleteTypesFromTargetAndSource();
+        createCategoriesCustomType(OLD_CATEGORY_CUSTOM_TYPE_KEY, Locale.ENGLISH, "anyName", CTP_TARGET_CLIENT);
+        createCategoriesCustomType(OLD_CATEGORY_CUSTOM_TYPE_KEY, Locale.ENGLISH, "anyName", CTP_SOURCE_CLIENT);
+    }
+
+    /**
+     * Deletes Categories and Types from source and target CTP projects, then it populates target CTP project with
+     * category test data.
+     */
+    @Before
+    public void setupTest() {
+        deleteAllCategories(CTP_TARGET_CLIENT);
+        deleteAllCategories(CTP_SOURCE_CLIENT);
```

Comment:
You delete all categories before each test method. Why are you explicitly call the deletion in some tests? Here for example:
https://github.com/commercetools/commercetools-sync-java/pull/72/files#diff-b70cdd276f9a92a4eabc591e1f9f9011R223

Reply:
because if you look at the line below this exact one you will find that categories are created always on every setup. However, for some specific cases (like the one you linked) I want to have no categories and create a specific category tree from scratch.